test/es-module/test-esm-loader-spawn-promisified.mjs

```diff
@@ -0,0 +1,288 @@
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+    assert.strictEqual(stdout, '');
+    assert.strictEqual(stderr, '');
+  });
+
+  it('can predetermine the format in the custom loader resolve hook', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      fixtures.fileURL('/es-module-loaders/hooks-custom.mjs'),
+      '--input-type=module',
+      '--eval',
+      `import assert from 'node:assert';
+      await import('esmHook/preknownFormat.pre')
+      .then((parsedModule) => {
+        assert.strictEqual(typeof parsedModule, 'object');
+        assert.strictEqual(parsedModule.default, 'hello world');
+      })`,
+    ]);
+
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+    assert.strictEqual(stdout, '');
+    assert.strictEqual(stderr, '');
+  });
+
+  it('can provide source for a nonexistent file', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      fixtures.fileURL('/es-module-loaders/hooks-custom.mjs'),
+      '--input-type=module',
+      '--eval',
+      `import assert from 'node:assert';
+      await import('esmHook/virtual.mjs')
+      .then((parsedModule) => {
+        assert.strictEqual(typeof parsedModule, 'object');
+        assert.strictEqual(typeof parsedModule.default, 'undefined');
+        assert.strictEqual(parsedModule.message, 'WOOHOO!');
+      })`,
+    ]);
+
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+    assert.strictEqual(stdout, '');
+    assert.strictEqual(stderr, '');
+  });
+
+  it('ensures that loaders have a separate context from userland', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      fixtures.fileURL('/es-module-loaders/hooks-custom.mjs'),
+      '--input-type=module',
+      '--eval',
+      `import assert from 'node:assert';
+      await import('${fixtures.fileURL('/es-modules/stateful.mjs')}')
+      .then(({ default: count }) => {
+        assert.strictEqual(count(), 1);
+      });`,
+    ]);
+
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+    assert.strictEqual(stdout, '');
+    assert.strictEqual(stderr, '');
+  });
+
+  it('ensures that user loaders are not bound to the internal loader', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      fixtures.fileURL('/es-module-loaders/loader-this-value-inside-hook-functions.mjs'),
+      '--input-type=module',
+      '--eval',
+      ';', // Actual test is inside the loader module.
+    ]);
+
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+    assert.strictEqual(stdout, '');
+    assert.strictEqual(stderr, '');
```

Comment:
Can you reverse the order here to simplify debugging? I find it much simpler to debug to start to check stderr/stdout as that where the information about what is wrong is, the code gives very little information.
```suggestion
    assert.strictEqual(stderr, '');
    assert.strictEqual(stdout, '');    
    assert.strictEqual(code, 0);
    assert.strictEqual(signal, null);
```

Reply:
Or assert all at once with an object?

Reply:
Using a object would work here because we are not using `assert.match`, but it wouldn't everywhere.

Reply:
Done